library/classes/Pharmacy.class.php

```diff
@@ -243,7 +247,8 @@ function pharmacies_factory($city = "", $sort = "ORDER BY name")
         $pharmacies = array();
         $sql = "SELECT p.id, a.city " .
             "FROM " . escape_table_name($p->_table) . " AS p " .
-            "INNER JOIN addresses AS a ON p.id = a.foreign_id " . $city . " " . add_escape_custom($sort);
+            "INNER JOIN addresses AS a ON p.id = a.foreign_id " . $city . " " . add_escape_custom($sort) .
+            " WHERE state = " . $this->state;
```

Comment:
can WHERE clause follow a ORDER clause?
Also will need to escape this:
```
" WHERE state = '" . add_escape_custom($this->state) . "'";
```

Reply:
done